common/trieForDirPath.go

```diff
@@ -0,0 +1,72 @@
+package common
+
+import (
+	"strings"
+)
+
+type TrieNode struct {
+	dirPathSeg             map[string]*TrieNode
+	status                 uint32
+	unregisteredButCreated bool
+	isEnd                  bool
+}
+
+type Trie struct {
+	Root *TrieNode
+}
+
+func NewTrie() *Trie {
+	return &Trie{
+		Root: &TrieNode{dirPathSeg: make(map[string]*TrieNode)},
+	}
+}
+
+// @brief InsertStatus inserts the status of the directory path in the trie
+func (t *Trie) InsertStatus(dirPath string, dirCreationStatus uint32) {
+	node := t.Root
+	segments := strings.Split(dirPath, "/")
+	for _, segment := range segments {
+		child, exists := node.dirPathSeg[segment]
+		if !exists {
+			child = &TrieNode{dirPathSeg: make(map[string]*TrieNode)}
+			node.dirPathSeg[segment] = child
+		}
+		node = child
+	}
+	node.status = dirCreationStatus
```

Comment:
This smells slightly. Is unregistered but created not in itself a status? Is it important to note that we created something, but that it is not a transfer we "own"?